### PR TITLE
Add mediafile.upload action + tests

### PR DESCRIPTION
**Pull request by reiterl**
Add tests. Mock the call of the Mediaservice in the tests.. So please have a close look at it.
Including mimetypes guessing, calculate has_inherited_access_groups and parent is_directory check.

### CODE REVIEW

File: openslides_backend/action/mediafile/upload.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            mimetype = mimetypes.guess_type(instance["instance"]["filename"])[0]
+            if mimetype is None:
+                raise ActionException(
+                    f"Cannot guess mimetype for {instance['instance']['filename']}."
+                )
+            self.upload_file(id_, file_, mimetype)
+        return dataset
+
+    def upload_file(self, id_: int, file_: str, mimetype: str) -> None:
+        ms = Mediaservice()
```
Comment (jsangmeister) [inline openslides_backend/action/mediafile/upload.py:70]: Please handle the media service analogous to the other services. Include it in the services container and pass it to actions in the constructor. I think it's better to keep this uniform. If you have questions about this, feel free to ask

File: tests/system/action/mediafile/test_upload.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                    ],
+                }
+            ],
+        )
+        self.assert_status_code(response, 200)
+        mediafile = self.get_model("mediafile/1")
+        assert mediafile.get("title") == "title_xXRGTLAJ"
+        assert mediafile.get("meeting_id") == 110
+        assert mediafile.get("filename") == "fn_jumbo.txt"
+        assert mediafile.get("file") is None
```
Comment (jsangmeister) [inline tests/system/action/mediafile/test_upload.py:32]: Please assert that `upload` was called. The mocked method should have a `assertCalledWith()` function.

File: tests/system/action/mediafile/test_upload.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                            "meeting_id": 110,
+                            "filename": "fn_jumbo.tasdde",
+                            "file": base64.b64encode(b"testtesttest").decode(),
+                        }
+                    ],
+                }
+            ],
+        )
+        self.assert_status_code(response, 400)
+        assert "Cannot guess mimetype for fn_jumbo.tasdde." in str(response.data)
```
Comment (jsangmeister) [inline tests/system/action/mediafile/test_upload.py:54]: Please assert that the mediafile does not exists and `upload` wasn't called.

**Review by jsangmeister — CHANGES_REQUESTED** [review]

**Comment by jsangmeister** [comment]
I will change this.

**Comment by jsangmeister** [comment]
Only wanted to clean up the interfaces a bit and whoops, 450 changes... well, most of it is pretty repetitive, but sorry that it got mixed into this PR :D

**Comment by jsangmeister** [comment]
@normanjaeckel if you want you can check this again. Merge it if you're fine with it

**Review by jsangmeister — APPROVED** [review]

File: openslides_backend/services/datastore/adapter.py [hunk 4]
```diff
@@ -182,7 +183,7 @@ def filter(
         collection: Collection,
         filter: Filter,
         mapped_fields: List[str] = None,
-        get_deleted_models: DeletedModelsBehaviour = DeletedModelsBehaviour.NO_DELETED,
+        get_deleted_models: DeletedModelsBehaviour = None,
```
Comment (normanjaeckel) [inline openslides_backend/services/datastore/adapter.py:186]: Please double check if this change is correct or a misstake.

File: openslides_backend/services/media/adapter.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    def __init__(self, media_url: str, logging: LoggingModule) -> None:
+        self.logger = logging.getLogger(__name__)
+        self.media_url = media_url
+
+    def upload(self, file: str, id: int, mimetype: str) -> Optional[Dict[str, str]]:
+        payload = {"file": file, "id": id, "mimetype": mimetype}
+        self.logger.debug(f"Starting upload of file '{file}'")
+        try:
+            response = requests.post(self.media_url, json=payload)
+        except Exception:
```
Comment (normanjaeckel) [inline openslides_backend/services/media/adapter.py:24]: Can you please use explicit excetions here? See https://github.com/OpenSlides/openslides-backend/blob/master/openslides_backend/services/datastore/http_engine.py#L49

**Review by normanjaeckel — APPROVED** [review]
Two small remarks

File: openslides_backend/services/datastore/adapter.py (hunk 4 above)
Comment (jsangmeister) [reply openslides_backend/services/datastore/adapter.py:186]: Yes, this is intentional: The way it is coded, the default is `NO_DELETED` anyway, and this change keeps it consistent with the other methods.

File: openslides_backend/services/media/adapter.py (hunk 5 above)
Comment (jsangmeister) [reply openslides_backend/services/media/adapter.py:24]: Done

**Pull request merged**